From ff5cdca154ed51d6dfdf67fec2049fb547bf3919 Mon Sep 17 00:00:00 2001
From: George Mileka <gmileka@microsoft.com>
Date: Wed, 23 Oct 2024 11:41:54 -0700
Subject: [PATCH] Update dracut to allow supressing user confirmation prompt
 when the liveos overlay is backed by memory.

Dracut allows the creation of a LiveOS using a read-only squashfs and an
read-write overlay on top.

If the read-write overlay is backed by a ram-disk, Dracut halts booting and
prompts the user to confirm whether to continue or not.

This interaction during the boot process is not desired in all cases.

This change introduces a new flag (rd.live.overlay.nouserconfirmprompt) that
when defined, it supresses the prompt and allows the boot process to continue
to completion without user interation.

There is no impact to existing configurations and their associated behavior.
Only when the new switch is explicitly define by the image build (as a kernel
parameter), the new behavior will take effect.
---
 modules.d/90dmsquash-live/dmsquash-live-root.sh | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/modules.d/90dmsquash-live/dmsquash-live-root.sh b/modules.d/90dmsquash-live/dmsquash-live-root.sh
index 43c57433..2e219e39 100755
--- a/modules.d/90dmsquash-live/dmsquash-live-root.sh
+++ b/modules.d/90dmsquash-live/dmsquash-live-root.sh
@@ -26,6 +26,10 @@ squash_image=$(getarg rd.live.squashimg)
 getargbool 0 rd.live.ram && live_ram="yes"
 getargbool 0 rd.live.overlay.reset && reset_overlay="yes"
 getargbool 0 rd.live.overlay.readonly && readonly_overlay="--readonly" || readonly_overlay=""
+# 'nouserconfirmprompt' is used to suppress a blocking prompt that asks for a
+# user confirmation before proceeding during boot time. This is to provide a
+# path for the image builder to boot it without user interaction.
+getargbool 0 rd.live.overlay.nouserconfirmprompt -d -y overlay_no_user_confirm_prompt && overlay_no_user_confirm_prompt="--noprompt" || overlay_no_user_confirm_prompt=""
 overlay=$(getarg rd.live.overlay)
 getargbool 0 rd.writable.fsimg && writable_fsimg="yes"
 overlay_size=$(getarg rd.live.overlay.size=)
@@ -219,7 +223,7 @@ do_live_overlay() {
     fi
 
     if [ -z "$setup" ] || [ -n "$readonly_overlay" ]; then
-        if [ -n "$setup" ]; then
+        if [ -n "$setup" -o -n "$overlay_no_user_confirm_prompt" ]; then
             warn "Using temporary overlay."
         elif [ -n "$devspec" ] && [ -n "$pathspec" ]; then
             [ -z "$m" ] \
-- 
2.34.1

